service/matching/taskListManager.go

```diff
@@ -243,23 +246,29 @@ func (c *taskListManagerImpl) completeTaskPoll(taskID int64) (ackLevel int64) {
 
 // Loads task from taskBuffer (which is populated from persistence) or from sync match to add task call
 func (c *taskListManagerImpl) getTask(ctx thrift.Context) (*getTaskResult, error) {
+	scope := metrics.MatchingTaskListMgrScope
 	timer := time.NewTimer(c.engine.longPollExpirationInterval)
 	defer timer.Stop()
 	select {
 	case task, ok := <-c.taskBuffer:
 		if !ok { // Task list getTasks pump is shutdown
 			return nil, errPumpClosed
```

Comment:
Should this also increment PollTimeoutCounter here? i am looking at that counter as total number of times the poll didn't give out a task.